### PR TITLE
Add functionality to upload images and pdfs

### PR DESCRIPTION
**Pull request by areejfnaqvi**
## Description

This PR corresponds to the following [Task](https://www.notion.so/conradlabshq/Upload-Image-PDF-Functionality-230a7d912e3148fa9933c56785ebc99c?pvs=4)

A user can now upload files (images and PDFs) to the chatbot.

The UI flicker issue has also been fixed as part of changes in this PR.

### CODE REVIEW

File: components/chat.tsx [hunk 1]
```diff
@@ -44,8 +44,10 @@ export function Chat({
   const [_, setNewChatId] = useLocalStorage('newChatId', id)
 
   useEffect(() => {
-    dispatch(setThreadId(''))
-    dispatch(removeMessages())
+    if (messages?.length !== 1) {
+      dispatch(setThreadId(''))
+      dispatch(removeMessages())
+    }
```
Comment (AwaisKamran) [inline components/chat.tsx:50]: What problem does this solve?

File: components/file-preview.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+const FilePreview: React.FC<FilePreviewProps> = ({
+  file,
+  previewUrl,
+  onRemove
+}) => {
+  if (!file) return null
+
+  const isImage = file.type.startsWith('image/')
+  const isPDF = file.type === 'application/pdf'
```
Comment (AwaisKamran) [inline components/file-preview.tsx]: Do we get these values by default? for image and pdfs
Didnt we try to restrict the types at HTML level

File: components/prompt-form.tsx [hunk 3]
```diff
@@ -45,16 +47,87 @@ export function PromptForm({
...
+    await openai.beta.threads.messages.create(currentThreadId, {
+      role: 'user',
+      content: value
+    })
+
+    const stream = await openai.beta.threads.runs.stream(currentThreadId, {
+      assistant_id: openAIAssistantId || '',
+      stream: true
+    })
+
```
Comment (AwaisKamran) [inline components/prompt-form.tsx]: It would be great if we encapsulate these methods into a function

File: components/prompt-form.tsx [hunk 4]
```diff
@@ -45,16 +47,87 @@ export function PromptForm({
...
+    let assistantResponse = ''
+    const newAssistantChatId = nanoid()
+
+    for await (const message of stream) {
+      if (
+        message.event === 'thread.message.delta' &&
```
Comment (AwaisKamran) [inline components/prompt-form.tsx]: The 'thread.message.delta' should be a constant

File: components/prompt-form.tsx [hunk 5]
```diff
@@ -45,16 +47,87 @@ export function PromptForm({
...
+        const text = (message.data.delta.content[0] as any).text.value
+          ? (message.data.delta.content[0] as any).text.value
+          : ''
+        assistantResponse += text
+
+        dispatch(
+          addMessage({
+            id: newAssistantChatId,
+            message: assistantResponse,
+            role: 'assistant'
```
Comment (AwaisKamran) [inline components/prompt-form.tsx]: Such trailing strings should also be a constant

File: components/prompt-form.tsx [hunk 6]
```diff
@@ -45,16 +47,87 @@ export function PromptForm({
...
+          })
+        )
+      }
+    }
+
+    const assistantMessage = {
+      id: newAssistantChatId,
+      message: assistantResponse,
+      role: 'assistant'
```
Comment (AwaisKamran) [inline components/prompt-form.tsx]: Similar comment to https://github.com/Conrad-Labs/nextjs-chatbot-ui-template/pull/3/files#r1860588476

File: components/prompt-form.tsx [hunk 7]
```diff
@@ -45,16 +47,87 @@ export function PromptForm({
...
+      : (chat.messages = [assistantMessage])
+    dispatch(addMessage(assistantMessage))
+
+    await saveChat(chat)
+    setIsAssistantRunning(false)
+
+    return {
+      id: messageId,
+      message: value,
+      role: 'user'
```
Comment (AwaisKamran) [inline components/prompt-form.tsx]: Similar comment https://github.com/Conrad-Labs/nextjs-chatbot-ui-template/pull/3/files#r1860588476

File: components/prompt-form.tsx [hunk 8]
```diff
@@ -84,44 +157,60 @@ export function PromptForm({
...
-      }
-      dispatch(addMessage(assistantMessage))
-      chat.messages
-        ? chat.messages.push(assistantMessage)
-        : (chat.messages = [assistantMessage])
-    } else {
-      console.error(run.status)
+    // Blur focus on mobile
+    if (window.innerWidth < 600) {
+      event.target['message']?.blur()
```
Comment (AwaisKamran) [inline components/prompt-form.tsx:190]: Was this an existing method?

File: components/chat.tsx (hunk 1 above)
Comment (areejfnaqvi) [reply components/chat.tsx:50]: It resets the redux store to initial state when we switch between chats. Without this, if we switched from chat A to chat B, we would see messages from chat A in chat B

File: components/prompt-form.tsx (hunk 8 above)
Comment (areejfnaqvi) [reply components/prompt-form.tsx:190]: Yes

**Review by AwaisKamran — APPROVED** [review]

**Pull request merged**